Cesium3DTilesSelection/src/TileUtilities.cpp

```diff
@@ -132,6 +141,107 @@ bool withinPolygons(
 
   return false;
 }
+
+bool outsidePolygons(
+    const BoundingVolume& boundingVolume,
+    const std::vector<CesiumGeospatial::CartographicPolygon>&
+        cartographicPolygons) noexcept {
+
+  std::optional<GlobeRectangle> maybeRectangle =
+      estimateGlobeRectangle(boundingVolume);
+  if (!maybeRectangle) {
+    return false;
+  }
+
+  return outsidePolygons(*maybeRectangle, cartographicPolygons);
+}
+
+bool outsidePolygons(
+    const CesiumGeospatial::GlobeRectangle& rectangle,
+    const std::vector<CesiumGeospatial::CartographicPolygon>&
+        cartographicPolygons) noexcept {
+
+  glm::dvec2 rectangleCorners[] = {
+      glm::dvec2(rectangle.getWest(), rectangle.getSouth()),
+      glm::dvec2(rectangle.getWest(), rectangle.getNorth()),
+      glm::dvec2(rectangle.getEast(), rectangle.getNorth()),
+      glm::dvec2(rectangle.getEast(), rectangle.getSouth())};
+
+  glm::dvec2 rectangleEdges[] = {
+      rectangleCorners[1] - rectangleCorners[0],
+      rectangleCorners[2] - rectangleCorners[1],
+      rectangleCorners[3] - rectangleCorners[2],
+      rectangleCorners[0] - rectangleCorners[3]};
+
+  // Iterate through all polygons.
+  for (size_t i = 0; i < cartographicPolygons.size(); ++i) {
+    const CartographicPolygon& selection = cartographicPolygons[i];
+
+    const std::optional<CesiumGeospatial::GlobeRectangle>&
+        polygonBoundingRectangle = selection.getBoundingRectangle();
+    if (!polygonBoundingRectangle ||
+        !rectangle.computeIntersection(*polygonBoundingRectangle)) {
+      continue;
+    }
+
+    const std::vector<glm::dvec2>& vertices = selection.getVertices();
+    const std::vector<uint32_t>& indices = selection.getIndices();
+
+    // First check if an arbitrary point on the bounding globe rectangle is
+    // inside the polygon.
+    for (size_t j = 2; j < indices.size(); j += 3) {
+      if (withinTriangle(
+              rectangleCorners[0],
+              vertices[indices[j - 2]],
+              vertices[indices[j - 1]],
+              vertices[indices[j]])) {
+        return false;
+      }
+    }
+
+    // Check if an arbitrary point on the polygon is in the globe rectangle.
```

Comment:
I think it would be better to check if the first polygon vertex is in the rectangle before checking if the first rectangle corner is in the polygon. Only because the former test will be much cheaper, and sometimes that cheaper test will save you from needing to do the more expensive one.